contracts/vesting/VestingWalletRecoveryLight.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: UNLICENSED
+// See Forta Network License: https://github.com/forta-network/forta-contracts/blob/master/LICENSE.md
+
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/utils/StorageSlot.sol";
+
+/**
+ * This contract is designed for recovery in case the beneficiary was lost.
+ *
+ * This contract replicates a storage layout that is compatible with VestingWallet, VestingWalletV1 and
+ * VestingWalletV2.
+ */
+contract VestingWalletRecoveryLight {
+    /// Storage
+    // Initializable
+    uint8 private _initialized;
+    bool private _initializing;
+    // ContextUpgradeable
+    uint256[50] private __gap_1;
+    // OwnableUpgradeable
+    address private _owner;
+    uint256[49] private __gap_2;
+    // UUPSUpgradeable
+    uint256[50] private __gap_3;
+    // ERC1967UpgradeUpgradeable
+    uint256[50] private __gap_4;
+    // VestingWallet / VestingWalletV1 / VestingWalletV2
+    mapping (address => uint256) private _released;
+    address private _beneficiary;
+    uint256 private _start;
+    uint256 private _cliff;
+    uint256 private _duration;
+    // VestingWalletV1 / VestingWalletV2
+    uint256[45] private __gap_5;
+    // VestingWalletV2
+    uint256 private historicalBalanceMin;
+    uint256[45] private __gap_6;
+
+    /// Constants and Events
+    // ERC1967UpgradeUpgradeable
+    bytes32 internal constant _IMPLEMENTATION_SLOT = 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc;
+    event Upgraded(address indexed implementation);
+    event BeneficiaryUpdate(address newBeneficiary);
+
+    function changeOwnerAndUpgrade(address newBeneficiary, address newImplementation) external {
```

Comment:
This is not the case, but I'm worried the function is not `changeOwnerAndUpgradeAndCall`. There's a case when the developer can upgrade to this implementation, call the `changeOwnerAndUpgrade` function, and then leave the new implementation uninitialized. That'd be ok if the new implementation is the previous implementation like here.

We should warn or imply this is intended to rollback. I'm leaning towards the latter:

```suggestion
    /// @dev The `previousImplementation` address MUST correspond to the real previous implementation.
    /// Otherwise, upgrading to this implementation contract and calling this function will leave the new implementation uninitialized.
    function changeBeneficiaryAndRollbackTo(address newBeneficiary, address previousImplementation) external {
```

Reply:
I like `...AndRollbackTo` in the name.

Reply:
Also +1 to renaming `changeOwner` to `changeBeneficiary`!